qt-bugs@ issue :
bugs.kde.org number :
applied: no
author: Dirk Mueller <mueller@kde.org>

fix compile errors with newer X.org

--- include/private/qt_x11_p.h
+++ include/private/qt_x11_p.h
@@ -86,7 +86,7 @@
 
 
 // the wacom tablet (currently just the IRIX version)
-#if defined (QT_TABLET_SUPPORT)
+#if defined (QT_TABLET_SUPPORT) && defined (QT_CLEAN_NAMESPACE)
 #  include <X11/extensions/XInput.h>
 #if defined (Q_OS_IRIX)
 #  include <wacom.h>  // wacom driver defines for IRIX (quite handy)
--- src/kernel/qtaddons_x11.cpp
+++ src/kernel/qtaddons_x11.cpp
@@ -22,6 +22,10 @@
  * PERFORMANCE OF THIS SOFTWARE.
  */
 
+#ifndef QT_CLEAN_NAMESPACE
+#define QT_CLEAN_NAMESPACE
+#endif
+
 #include "qt_x11_p.h"
 
 #if !defined(QT_NO_XFTFREETYPE) && !defined(QT_XFT2)
--- src/kernel/qt_x11_p.h
+++ src/kernel/qt_x11_p.h
@@ -86,7 +86,7 @@
 
 
 // the wacom tablet (currently just the IRIX version)
-#if defined (QT_TABLET_SUPPORT)
+#if defined (QT_TABLET_SUPPORT) && defined (QT_CLEAN_NAMESPACE)
 #  include <X11/extensions/XInput.h>
 #if defined (Q_OS_IRIX)
 #  include <wacom.h>  // wacom driver defines for IRIX (quite handy)
--- src/kernel/qwidget_x11.cpp
+++ src/kernel/qwidget_x11.cpp
@@ -38,6 +38,10 @@
 **
 **********************************************************************/
 
+#ifndef QT_CLEAN_NAMESPACE
+#define QT_CLEAN_NAMESPACE
+#endif
+
 #include "qapplication.h"
 #include "qapplication_p.h"
 #include "qnamespace.h"
